### PR TITLE
allow get /jobs to query on subscription_id

### PR DESCRIPTION
**Pull request by Jlrine2**

### CODE REVIEW

File: tests/test_api/test_list_jobs.py [hunk 1]
```diff
@@ -151,6 +151,26 @@ def test_list_jobs_date_start_and_end(client, tables):
...
+
+    login(client)
+    response = client.get(JOBS_URI, query_string={'subscription_id': '9b02d992-e21e-4e2f-9310-5dd469be2708'})
+    assert response.status_code == HTTPStatus.OK
+    assert list_have_same_elements(response.json['jobs'], items[:2])
+
+    response = client.get(JOBS_URI, query_string={'subscription_id': '9b02d992-e21e-4e2f-9310-5dd469be2700'})
+    assert response.status_code == HTTPStatus.OK
+    assert response.json['jobs'] == [items[2]]
+
```
Comment (asjohnston-asf) [inline tests/test_api/test_list_jobs.py:172]: add a test searching for a non-existent subscription_id and confirm we get an empty list back?

**Review by jhkennedy — APPROVED** [review]

**Review by jhkennedy — APPROVED** [review]
